### DISCUSSION TITLE
Option to skip Caching

### DISCUSSION THREAD

**Discussion by grizzlybearer** [openfga/community#280] [Ideas]
OpenFGA has a (server in-memory) [caching feature](https://github.com/openfga/openfga/blob/819402f231b655a6d787adf36a735b7fe676554c/cmd/run/run.go#L194) that applies to Check and ListObjects APIs by caching the sub-problems.

However, there can be multiple use-cases running on the same set of OpenFGA servers, some of which may benefit from caching and some that may not tolerate any caching. There is no currently no way of skipping caching based on any condition.

1 idea could be to provide an optional `skipCaching` flag in the Check and ListObjects API to overcome this.

We have such use-cases currently running in production, and are unable to toggle caching due to this limitation.

**Comment by aaguiarz** [comment]
Hi @grizzlybearer !

Do you think this feature would address the issue you have https://github.com/openfga/roadmap/issues/54?

**Comment by aaguiarz** [comment]
This was shipped! https://github.com/openfga/openfga/pull/1764
